### PR TITLE
1.x: No more need to convert Singles to Observables for Single.zip()

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
PTAL

### CODE REVIEW

**Comment by akarnokd** [comment]
:+1:

File: src/main/java/rx/Single.java [hunk 1]
```diff
@@ -966,8 +966,13 @@ public void onError(Throwable error) {
      * @return a Single that emits the zipped results
      * @see <a href="http://reactivex.io/documentation/operators/zip.html">ReactiveX operators documentation: Zip</a>
      */
-    public final static <T1, T2, R> Single<R> zip(Single<? extends T1> o1, Single<? extends T2> o2, final Func2<? super T1, ? super T2, ? extends R> zipFunction) {
-        return just(new Observable<?>[] { asObservable(o1), asObservable(o2) }).lift(new OperatorZip<R>(zipFunction));
+    public static <T1, T2, R> Single<R> zip(Single<? extends T1> s1, Single<? extends T2> s2, final Func2<? super T1, ? super T2, ? extends R> zipFunction) {
```
Comment (zsxwing) [inline src/main/java/rx/Single.java]: nit: please add `final` back.

**Comment by zsxwing** [comment]
Could you add `final` back? Otherwise, LGTM

File: src/main/java/rx/Single.java (hunk 1 above)
Comment (JakeWharton) [inline src/main/java/rx/Single.java]: or remove `final` from all static methods in another PR

Comment (akarnokd) [inline src/main/java/rx/Single.java]: I'm for the removal as well, it's redundant. (I also remember an argument about it some time back.)

**Comment by artem-zinnatullin** [comment]
Sorry, was busy… @zsxwing added `final` back, will send separate PR to remove them everywhere once this merged.

**Comment by zsxwing** [comment]
:+1:

**Pull request merged**